### PR TITLE
Feature/frontend tasks per user

### PR DESCRIPTION
**Pull request by valfink**

### CODE REVIEW

File: frontend/src/component/DayGallery.tsx [hunk 1]
```diff
@@ -3,13 +3,15 @@ import {Task} from "../model/Task";
...
+import {User} from "../model/User";
 
 
 
 type Props = {
     year: string
     month: string
-    day: string
+    day: string,
+    user: User | null
```
Comment (MoinMarcell) [inline frontend/src/component/DayGallery.tsx:14]: use undefined instead of null

File: frontend/src/component/TaskCard.tsx [hunk 2]
```diff
@@ -2,9 +2,11 @@ import {Task} from "../model/Task";
 import React from "react";
 import "./TaskCard.css";
 import {Link} from "react-router-dom";
+import {User} from "../model/User";
 
 type Props={
-    task:Task
+    task: Task,
+    user: User | null
```
Comment (MoinMarcell) [inline frontend/src/component/TaskCard.tsx:9]: same here, use undefined instead of null

File: frontend/src/component/TaskCard.tsx [hunk 3]
```diff
@@ -15,7 +17,7 @@ export default function TaskCard(props:Props){
            <br/>
            <Link to={"/tasks/"+props.task.id}>Details</Link>
            <br/>
-           <Link to={`/tasks/${props.task.id}/update`}>Edit</Link>
+           {props.user && props.user.id === props.task.userId ? <Link to={`/tasks/${props.task.id}/update`}>Edit</Link> : "\u00A0" }
        </div>
    )
 }
```
Comment (MoinMarcell) [inline frontend/src/component/TaskCard.tsx:23]: no new line at eof

File: frontend/src/model/User.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+export type User = {
+    id: string;
+    username: string;
+    role: string;
+}
```
Comment (MoinMarcell) [inline frontend/src/model/User.ts:5]: no new line at eof

File: frontend/src/page/TasksGallery.tsx [hunk 5]
```diff
@@ -10,7 +10,7 @@ type Props = {
 export default function TasksGallery(props: Props) {
     const user = useAuth(true)
     const taskCards = props.tasks.map((task) => {
-            return <TaskCard key={task.id} task={task}/>
+            return <TaskCard key={task.id} task={task} user={user}/>
         }
     )
     return !user ? null : (
```
Comment (MoinMarcell) [inline frontend/src/page/TasksGallery.tsx:16]: null is no good practice. just return an empty string here
```suggestion
    return !user ? '' : (
```

**Review by lostspezi — APPROVED** [review]
lgtm, suggestions down below :)

**Pull request merged**